File: 0242-valid-anagram/0242-valid-anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length) return false;
        if (s == t) return true;
        
        List<char> list = new List<char>(s);
        for(int i=0; i<t.Length; i++)
        {
            if(list.Contains(t[i]))
            {
                list.Remove(t[i]);
            }
            else
            {
                return false;
            }
        }
        return list.Count > 0 ? false : true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        countS, countT = {}, {}

        #create the hashMaps 
        for i in range(len(s)):
            countS[s[i]] = countS.get(s[i], 0) + 1
            countT[t[i]] = countT.get(t[i], 0) + 1

        #compare the counts of each character in the hashMaps
        for c in countS:
            if countS[c] != countT.get(c, 0):
                return False
        
        return True